swift/ql/src/queries/Security/CWE-311/CleartextTransmission.ql

```diff
@@ -54,6 +54,25 @@ class Url extends Transmitted {
   }
 }
 
+/**
+ * An `Expr` that transmitted through the Alamofire library.
+ */
+class AlamofireTransmitted extends Transmitted {
+  AlamofireTransmitted() {
+    // sinks are the first argument containing the URL, and the `parameters`
+    // and `headers` arguments to appropriate methods of `Session`.
+    exists(CallExpr call, string fName |
+      call.getStaticTarget().(MethodDecl).hasQualifiedName("Session", fName) and
+      fName.regexpMatch("(request|streamRequest|download)\\(.*") and
```

Comment:
I'm not sure if `matches` is more performant than `regexpMatch` (I think it is). If that's the case, this might be better:

```suggestion
      fName.matches(["request", "streamRequest", "download"] + "(%") and
```

Reply:
For me this was just over the threshold where its cleaner (and more efficient???) to just use a regexp.  I'm open to persuasion though (particularly on the 'more efficient' part because its only a weak intuition).

Reply:
True.  I am slightly curious which is faster (`match` is generally faster than `regexpMatch` but there's a sense in which this is three uses of `match`).  But lets not worry about it now.